### ISSUE TITLE
[FrameworkBundle][Routing] deprecation error when decorating routing.loader

### ISSUE THREAD

**Issue by kbond** [symfony/symfony#24886]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4-BETA

When decorating `routing.loader` in a compiler pass I get the following deprecation error when the container is built:

```
User Deprecated: The "routing.loader" service is private, getting it from the container is deprecated since Symfony 3.2 and will fail in 4.0. You should either make the service public, or stop using the container directly and use dependency injection instead.
```

I can't figure out why this is happening.

Reference: https://github.com/umpirsky/UmpirskyI18nRoutingBundle/issues/17
CompilerPass: https://github.com/umpirsky/UmpirskyI18nRoutingBundle/blob/a69bd4dcc901c3459aba20f8e586613a5758d05b/src/DependencyInjection/Compiler/OverrideRoutingCompilerPass.php

![profiler](https://user-images.githubusercontent.com/127811/32618918-0d6df242-c547-11e7-94d2-30562215dee0.png)

**Comment by sroze** [comment]
@kbond which beta is that?

**Comment by sroze** [comment]
Well, actually it's clearly the `UmpirskyI18nRoutingBundle` relying on a private service... I'm not sure why it is doing that but it's very likely it can be re-written on their side to only use the router and tagging their resources as route loader I guess.

**Comment by ogizanagi** [comment]
The `router.loader` service still is public (see #24738), but the pass decorates it and create an alias:

https://github.com/umpirsky/UmpirskyI18nRoutingBundle/blob/a69bd4dcc901c3459aba20f8e586613a5758d05b/src/DependencyInjection/Compiler/OverrideRoutingCompilerPass.php#L27

This alias should be explicitly marked as public since #24104, otherwise it'll be considered private.

**Comment by xabbuh** [comment]
closing here as this is indeed not a bug as @ogizanagi explained

**Comment by stof** [comment]
Well, the simpler solution for the UmpirskyI18nRoutingBundle would be to use the service decoration feature of Symfony, which is available since Symfony 2.5, instead of duplicating it.
This way, it would not have to deal with the changes required in the implementation of decoration for newer versions of Symfony (and would not make assumptions about the visibility of core services)

**Comment by kbond** [comment]
Thanks everyone, service decoration solved my issue.
